R/utils.R

```diff
@@ -237,15 +237,23 @@ rtoot_ask <- function(prompt = "enter authorization code: ", pass = TRUE, check_
     return(passFun(prompt = prompt))
 }
 
+handle_id <- function(x) {
+    ## Convert x to snowflake id if it is POSIXct
+    if (is (x, "POSIXct")) {
+        return(as.numeric(x)*(2^16))*1000)
```

Comment:
Sorry, my bad:

```r
handle_id <- function(x) {
    ## Convert x to snowflake id if it is POSIXct
    if (is(x, "POSIXct")) {
        return(as.numeric(x) * (2^16) * 1000)
    }
    x
}
```

It is also worth mentioning that there are bitshift operators in R.

```r
bitwShiftL(10, 16) == (10 * 2^16)
```

Reply:
I had trouble getting bitwShiftL to work because the mastodon ids are bigger then the max integer (at least in my testing).

Reply:
Ah, yes, 64-bit integer.